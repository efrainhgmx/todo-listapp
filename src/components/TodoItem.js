import React from 'react';

export const TodoItem = ({ todo, toggleTodo }) => {
    const {id, task, completed } = todo;

    const handleTodoClick = () => {
        toggleTodo(id);
    }

    return (
        <li className="listItem">
            <input type="checkbox" id={ id } checked={ completed }  onChange={ handleTodoClick }/>
            <label htmlFor={ id }>{ task }</label>
        </li>
    )
}
