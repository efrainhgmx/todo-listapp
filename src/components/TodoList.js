import React from 'react';

export const TodoList = ({ todos }) => {
    return (
        <ul>
            { todos.map((todo) => (
                <li>Tarea</li>
            )) }
        </ul>
    )
}
