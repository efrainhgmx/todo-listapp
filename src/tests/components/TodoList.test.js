import React  from 'react';
import { v4 as uuidv4 } from 'uuid';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { TodoList } from '../../components/TodoList';

describe('Test on <TodoList /> components', () => {
    test('Should be have the same lenght of task list', () => {
        const todos = [{id: uuidv4(), task: "Tarea 1", completed: false },{id: uuidv4(), task: "Tarea 2", completed: false}];
        const wrapper = shallow( <TodoList todos={ todos }/>);

        expect( wrapper ).toMatchSnapshot();
        expect( todos.length ).toBe(2);
    });
    
})
