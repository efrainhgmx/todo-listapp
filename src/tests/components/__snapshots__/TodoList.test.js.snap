// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <TodoList /> components Should be have the same lenght of task list 1`] = `
<ul>
  <TodoItem
    key="8dd399f2-931c-46c8-9bfa-cd14d078b6f5"
    todo={
      Object {
        "completed": false,
        "id": "8dd399f2-931c-46c8-9bfa-cd14d078b6f5",
        "task": "Tarea 1",
      }
    }
  />
  <TodoItem
    key="f7a32b6a-df10-4931-b597-42d86f453715"
    todo={
      Object {
        "completed": false,
        "id": "f7a32b6a-df10-4931-b597-42d86f453715",
        "task": "Tarea 2",
      }
    }
  />
</ul>
`;
